packages/cdk-routing-split/src/add/handler.ts

```diff
@@ -2,24 +2,43 @@ import { APIGatewayEvent, Context } from 'aws-lambda';
 import { createLogger, Logger, serializeError } from '@daysmart/aws-lambda-logger';
 import { action } from './action';
 import { HttpError } from '../http-error';
+import { AddRequest, Key } from './interface';
 
-export const handler = async (event: APIGatewayEvent, context: Context): Promise<any> => {
+export const add = async (event: APIGatewayEvent, context: Context): Promise<any> => {
     let logger!: Logger;
 
     try {
         logger = createLogger(process.env.DEBUG === 'true', context.awsRequestId);
         logger.debug('add event', { event });
 
+        const request: AddRequest = JSON.parse(event.body as string);
+        validateRequest(request);
+
+        await action(request);
+
         return {
             statusCode: 200,
-            body: action(),
+            body: 'Success',
         };
     } catch (error: any) {
         logger?.error('handler_error', { logError: serializeError(error) });
+
         if (error instanceof HttpError) {
             return { statusCode: error.statusCode, body: error.message };
         } else {
             return { statusCode: 500, body: error.message };
         }
     }
 };
+
+const validateRequest = (request: AddRequest): void => {
+    const keyList: string[] = Object.values(Key);
```

Comment:
this enum isn't really being used as an enum but a list. I would instead just make it a list and either hard code it here, or in the handler and pass it down (id probably just hard code it right here directly in the function since its not used elsewhere).

Reply:
It is actually used in test, but I can make value in test to be string, like used would use it.